mypy/test/testpythoneval.py

```diff
@@ -80,21 +75,23 @@ def test_python_evaluation(testcase: DataDrivenTestCase) -> None:
 
     # Write the program to a file.
     program = '_' + testcase.name + '.py'
-    mypy_cmdline.append(program)
     program_path = os.path.join(test_temp_dir, program)
+    mypy_cmdline.append(program_path)
     with open(program_path, 'w') as file:
         for s in testcase.input:
             file.write('{}\n'.format(s))
+    output = []
     # Type check the program.
-    # This uses the same PYTHONPATH as the current process.
-    returncode, out = run(mypy_cmdline)
+    out, err, returncode = api.run(mypy_cmdline)
+    output.extend([s.rstrip('\r\n').lstrip(test_temp_dir + os.sep)
```

Comment:
Looks like you're misunderstanding how `lstrip()` works. It removes characters from the left as long as they are in the argument, which is surely not what you meant. (Ditto for `rstrip()` but in this case it's harmless.)

Reply:
Ah, you are right. Apologies. The rstrip should be fine (removing trailing newlines is what I want), but the lstrip I can replace with a slice. Thanks!